hrc_speech_prediction/tree.py

```diff
@@ -0,0 +1,248 @@
+import os
+import argparse
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.externals import joblib
+
+from hrc_speech_prediction import features
+from hrc_speech_prediction.data import (TrainData,
+                                        TRAIN_PARTICIPANTS, ALL_ACTIONS)
+
+
+parser = argparse.ArgumentParser("Train and evaluate classifier")
+parser.add_argument(
+    'path', help='path to the experiment data', default=os.path.curdir)
+
+
+class Tree(object):
```

Comment:
So, actually it seems that this class is not so much about the tree. A small part of it is about having a root and maintaining a pointer to the current node but the rest of it is about merging the speech and context model.

What you want to have is something called a _DualModel_ or a _MergedModel_ or whatever that takes care of that and that has two attributes, one for the speech model and one for the context model. Currently the context model would be a `Tree` but you might want to rename `TreeContextModel`. Ideally try to keep the function calls to the speech and context model similar.